### ISSUE TITLE
zip-archive-0.3.2 is failing tests on nightly-2018-01-17

### ISSUE THREAD

**Issue by cdornan** [jgm/zip-archive#38]
zip-archive-0.3.2 is failing tests on nightly-2018-01-17 with the following output:

```
[1 of 1] Compiling Main             ( tests/test-zip-archive.hs, dist/build/test-zip-archive/test-zip-archive-tmp/Main.o )

tests/test-zip-archive.hs:12:1: warning: [-Wunused-imports]
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
   |
12 | import Control.Applicative
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
Linking dist/build/test-zip-archive/test-zip-archive ...
> /tmp/stackage-build12/zip-archive-0.3.2$ dist/build/test-zip-archive/test-zip-archive
^MCases: 8  Tried: 0  Errors: 0  Failures: 0^MCases: 8  Tried: 1  Errors: 0  Failures: 0^MCases: 8  Tried: 2  Errors: 0  Failures: 0^M                                          ^M### Error in:   2
tests/test_dir_with_symlinks: getSymbolicLinkStatus: does not exist (No such file or directory)
^MCases: 8  Tried: 3  Errors: 1  Failures: 0^MCases: 8  Tried: 4  Errors: 1  Failures: 0  adding: LICENSE (deflated 46%)
  adding: src/ (stored 0%)
  adding: LICENSE (deflated 46%)
  adding: src/ (stored 0%)
  adding: src/Codec/ (stored 0%)
  adding: src/Codec/Archive/ (stored 0%)
  adding: src/Codec/Archive/Zip.hs (deflated 74%)
^M                                          ^M### Error in:   4
tests/test_dir_with_symlinks: getSymbolicLinkStatus: does not exist (No such file or directory)
^MCases: 8  Tried: 5  Errors: 2  Failures: 0^MCases: 8  Tried: 6  Errors: 2  Failures: 0  creating: test-zip-archive.1078/dir1
extracting: test-zip-archive.1078/dir1/hi
  creating: test-zip-archive.1078/dir1/dir2
 inflating: test-zip-archive.1078/dir1/dir2/hello
^MCases: 8  Tried: 7  Errors: 2  Failures: 0  creating: test-zip-archive.1078/dir3
extracting: test-zip-archive.1078/dir3/hi
^M                                          ^MCases: 8  Tried: 8  Errors: 2  Failures: 0
```
We have added it to the `expected-test-failures` list &mdash; please let us know when you think the tests should pass.

**Comment by jgm** [comment]
@blender  The tests you added required some source files which were not placed in extra-source-files in the cabal file.

I've updated the travis script to catch problems like this in the future.

Unfortunately in this case we can't just include the source files, because one is a symbolic link to a directory. Instead, we have to build the directory programatically.  I've made a start on this in commit e6fa38886e1cdcba7ef90d4f5e648acbed6f4a65, but a test fails, for reasons I haven't figured out yet.

**Comment by jgm** [comment]
Fixed by #39 and 0.3.2.1 release.

**Comment by DanBurton** [comment]
The new version of your test suite requires `directory >= 1.3.1`, but we peg the version of this library to the version shipped with ghc: `directory-1.3.0.2`. I've added zip-archive to the `skipped-tests` section now, since including this test suite is incompatible with the build plan.

**Comment by blender** [comment]
@jgm  I think `directory-1.3.0.2` version would do

**Comment by jgm** [comment]
According to the changelog we don't get createFileLink
or createDirectoryLink til 1.3.1.0.

+++ Tommaso Piazza [Jan 22 18 14:23 ]:

**Comment by blender** [comment]
@jgm I can change that to use Posix functions

**Comment by jgm** [comment]
@DanBurton I just released 0.3.2.2 which only requires directory >= 1.3, so you should be able to remove us from skipped-tests.

**Comment by DanBurton** [comment]
Great, thanks!
